src/packagedcode/nuget.py

```diff
@@ -179,3 +180,93 @@ def parse(cls, location, package_only=False):
         )
         yield models.PackageData.from_data(package_data, package_only)
 
+class NugetPackagesLockHandler(models.DatafileHandler):
+    datasource_id = 'nuget_packages_lock'
+    path_patterns = ('packages.lock.json',)
+    default_package_type = 'nuget'
+    description = 'NuGet packages.lock.json file'
+    documentation_url = 'https://docs.microsoft.com/en-us/nuget/reference/packages-lock-file'
+
+    @classmethod
+    def is_datafile(cls, location, filetypes=..., _bare_filename=False):
+        return super().is_datafile(location, filetypes, _bare_filename) and location.endswith('.lock.json')
+    
+    @staticmethod
+    def get_dependencies(package_info, scope):
+        dependencies = []
+        for dep in package_info.get('dependencies', {}):
+            dependency = models.DependentPackage(
+                purl=str(PackageURL(type='nuget', name=dep, version=package_info["dependencies"][dep])),
+                extracted_requirement=package_info["dependencies"][dep],
+                is_resolved=True,
+                scope=scope,
+                is_optional=False,
+                is_runtime=True,
+                is_direct=False,
```

Comment:
```suggestion
                is_direct=True,
```
`is_direct: True` is not only for the direct dependencies of the main root package, but also for all direct dependency relationships of transitive dependencies.